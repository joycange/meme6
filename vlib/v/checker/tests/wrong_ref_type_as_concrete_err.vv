import datatypes

[heap]
struct User {
	name string
}

fn main() {
	mut dl := datatypes.DoublyLinkedList[&User]{}
	println(dl)
}
