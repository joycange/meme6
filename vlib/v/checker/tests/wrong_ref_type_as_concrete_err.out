vlib/v/checker/tests/wrong_ref_type_as_concrete_err.vv:9:39: error: cannot use &Type as generic concrete type, declare the function parameter or struct member as &T instead
    7 | 
    8 | fn main() {
    9 |     mut dl := datatypes.DoublyLinkedList[&User]{}
      |                                          ^
   10 |     println(dl)
   11 | }
